lib/seahorn/BvOpsem2WideMemMgr.cc

```diff
@@ -314,10 +316,7 @@ class WideMemManager : public OpSemMemManagerBase {
       return ptrAdd(ptr, _offset.get_si());
     }
     auto address = m_ctx.alu().doAdd(ptr.getRaw(), offset, ptrSzInBits());
-    auto new_size = m_ctx.alu().doSub(
-        ptr.getSize(),
-        m_ctx.alu().doZext(offset, g_slotBitWidth, m_main.wordSzInBits()),
-        g_slotBitWidth);
+    auto new_size = m_ctx.alu().doSub(ptr.getSize(), offset, g_slotBitWidth);
```

Comment:
if `ptrSzInBits()  != g_slotBitWidth` then this will fail because `offset` must be of one width or the other.
Should use `sext` to sign extended offset (offset can be negative) or trunc offset to fit the other bit-width.

I believe offset must be same bitwidth as `ptrSzInBits()` so it might require shrinking of `g_slotBitWidth` is smaller.